### ISSUE TITLE
Write up WWDC question re: snapshot testing

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#432]

**Comment by finestructure** [comment]
Request submitted

> We’re running webpage snapshot tests for our Server Side Swift Vapor service but none of the snapshots made with WKWebView’s load method render css or load images. It appears that baseURL does not work when run via an SPM target. Sandboxing issue? On macOS app targets this works by adding a network entitlement.
> https://github.com/pointfreeco/swift-snapshot-testing/issues/345

**Comment by Sherlouk** [comment]
There's clearly some absolute sandbox schenangians happening here. It seems WKWebView really doesn't like accessing file URLs *especially* files which are not within the same directory as the source file.

I tried using this method below where `url` is the HTML body saved to a local file where I get an explicit sandbox error.
> webView.loadFileURL(url, allowingReadAccessTo: baseURL)

> 2020-06-23 21:11:56.982392+0100 xctest[68906:1718182] [Process] 0x102816420 - [pageProxyID=5, webPageID=6, PID=68910] WebPageProxy::Ignoring request to load this main resource because it is outside the sandbox

I have managed to get it so the HTML file without any modifications works when loaded into Chrome but just with the sandbox of the test runner it seems to be really getting stuck no matter what I try (even tried some private APIs for `allowFileAccessFromFileURLs` and `allowUniversalAccessFromFileURLs` but didn't make much progress...)

One thing we _could_ do is inject the CSS in as an in-head <style> but this would be a lot of "test only" code which I'm not really keen on.

I've tried all sorts of attempts to try and save the HTML and resources into a directory and load that (code was horrible) but even with that I couldn't get it working. I sense the only way this would work is if the resources are committed into the build itself (the file should be accessible via Bundle.main for it to be seen by WKWebView)

It may be more possible with the new SPM Resource capabilities to add the public resources to the test target via the Package.swift which could resolve that issue? Might be one to try out!
